### ISSUE TITLE
Create Form with pre-defined set of fields

### ISSUE THREAD

**Issue by lingster** [labd/wagtailstreamforms#156]
Hi, 
Is it possible to create a form which is programmatically pre-defined with a set of fields. I'd like to ensure that my forms always has at least 1 input text box, but I'm stuggling to see how to add this.

**Comment by ababic** [comment]
@lingster you could do this by overriding the `base_form_class` on the `Form` model, and setting an initial value for fields in the `__init__` method, but it would be nicer to be able to swap out the form model completely, then you could simply override `__init__` on your custom model.

**Comment by lingster** [comment]
@ababic : indeed this is roughly what I have tried: 

```
class MyPageForm(WagtailAdminModelForm):
    def clean(self):
       # my custom code 

class MyForm(AbstractForm):
    base_form_class = MyPageForm
    def __init__(self):
      ....
    
@modeladmin_register
class MyFormModelAdmin(FormModelAdmin):
    model = MyForm
```
The trouble with this method is that we will now have additional table created to store MyForm. I'm just looking for a cleaner way to implement this, could you do something like `Form.base_form_class = MyPageForm` instead?

**Comment by ababic** [comment]
> could you do something like Form.base_form_class = MyPageForm instead?

This is the approach I have taken in a recent project.

**Comment by lingster** [comment]
Yes, that does indeed do the trick. I can add some documentation for this, if it will be useful for future users?
